Gruntfile.js

```diff
@@ -93,7 +93,7 @@ module.exports = grunt => {
 
     let user = grunt.config('gitinfo.local.branch.current.currentUser');
     if (user) {
-      user = user.substr(0, user.indexOf(' ')).toLowerCase();
+      user = user.substring(0, user.indexOf(' ')).toLowerCase();
```

Comment:
im not clear why we are using substring some places and slice some places.

Reply:
`substr()`, `slice()` and `substring()` are very similar but act differently in [some situations](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#differences_between_substring_and_slice), especially if the second parameter is negative.
If the second parameter is normally positve but can be negative you most likely want to use substring() as slice() would produce a different behaviour than substr().

Example:
```
"0123456789".substr(0, -1) == ""
"0123456789".slice(0, -1) == "012345678"
"0123456789".substring(0, -1) == ""
```

Sometimes you want the `slice()` behaviour but not in the cases in the code here.

It would be possible to use `slice()` everywhere if you add a check, like

`      user = user.substring(0, user.indexOf(' ') > 0 ? user.indexOf(' ') : 0).toLowerCase();`

Using substring() does look a bit nicer though if you don't mind mixing the two.

Reply:
ah sorry didnt read this close enough didn't realize the second param was indexOf, makes perfect sense then.